pkg/scheduler/queue/tenant_querier_assignment.go

```diff
@@ -249,20 +251,19 @@ func (tqa *tenantQuerierAssignments) setup(dequeueArgs *DequeueArgs) {
 // Note that because we use the shared  tenantIDOrder and tenantOrderIndex to manage the queue, we functionally
 // ignore each Node's individual queueOrder and queuePosition.
 func (tqa *tenantQuerierAssignments) dequeueSelectNode(node *Node) *Node {
+	if node.isLeaf() || len(node.queueMap) == 0 {
+		return node
+	}
+
 	// can't get a tenant if no querier set
 	if tqa.currentQuerier == "" {
 		return nil
 	}
 
-	// advance queue position for dequeue
+	// tenantOrderIndex is set to the _last_ tenant we dequeued from; advance queue position for dequeue
```

Comment:
"is set to" makes this a little unclear - it is set to before this line or this is the line that sets it?
"starts at" or "was set to" would help